workdir/copy/link/build.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+comp=""
+plat=""
+arch=""
+fs=""
+extra=""
+
+name=$(basename "$0" .sh)
+
+echo "$name" | grep "^clang-" > /dev/null 2>&1
+if test $? -eq 0; then
+    comp="clang"
+else
+    echo "$name" | grep "^gcc-" > /dev/null 2>&1
+    if test $? -eq 0; then
+        comp="gcc"
+    else
+        echo "Unknown compiler" 1>&2
+        exit 1
+    fi
+fi
```

Comment:
Maybe we can also get the compiler version (if any) here (i.e. if the filename is `gcc-11-build-qemu-x86_64`, `$comp` will be `gcc-11`, if the filename is `gcc-build-qemu-x86_64`, `$comp` will be `gcc`)?

Something like
```
echo "$name" | grep -E "^gcc-[0-9]+" && comp=$(echo "$name" | cut -d- -f1-2) || comp=$(echo "$name" | cut -d- -f1)
```

Then, if we just want to test for a specific compiler version, we can copy and rename the script from the application directory we are testing from.

Reply:
Nice, yes, I like that.